aiida/cmdline/commands/cmd_profile.py

```diff
@@ -133,69 +135,49 @@ def profile_setdefault(profile):
 
 
 @verdi_profile.command('delete')
-@options.FORCE(help='to skip questions and warnings about loss of data')
-@click.option(
-    '--include-config/--skip-config',
-    default=True,
-    show_default=True,
-    help='Include deletion of entry in configuration file.'
-)
+@options.FORCE(help='Skip any prompts for confirmation.')
 @click.option(
-    '--include-db/--skip-db',
-    'include_database',
-    default=True,
-    show_default=True,
-    help='Include deletion of associated database.'
-)
-@click.option(
-    '--include-db-user/--skip-db-user',
-    'include_database_user',
+    '--delete-storage/--no-delete-storage',
     default=False,
     show_default=True,
-    help='Include deletion of associated database user.'
-)
-@click.option(
-    '--include-repository/--skip-repository',
-    default=True,
-    show_default=True,
-    help='Include deletion of associated file repository.'
+    help='Whether to delete the storage with all its data or not.'
 )
 @arguments.PROFILES(required=True)
-def profile_delete(force, include_config, include_database, include_database_user, include_repository, profiles):
+@click.pass_context
+def profile_delete(ctx, force, delete_storage, profiles):
     """Delete one or more profiles.
 
     The PROFILES argument takes one or multiple profile names that will be deleted. Deletion here means that the profile
-    will be removed including its file repository and database. The various options can be used to control which parts
-    of the profile are deleted.
+    will be removed from the config file. If ``--delete-storage`` is specified, the storage containing all data is also
+    deleted.
     """
-    if not include_config:
-        echo.echo_deprecated('the `--skip-config` option is deprecated and is no longer respected.')
+    default_profile_removed: str | None = None
+    default_profile: str | None = ctx.obj.config.default_profile_name
 
     for profile in profiles:
+        suffix = 'including' if delete_storage else 'without'
+        echo.echo_warning(f'Deleting profile `{profile.name}`, {suffix} the storage.')
+        echo.echo_warning('This operation cannot be undone, ', nl=False)
```

Comment:
Maybe move this sentence inside the `click.confirm`, since I think it only makes sense there? Now you get

```
❯ verdi profile delete sqlitedos -f --delete-storage
Warning: Deleting profile `sqlitedos`, including the storage.
Warning: This operation cannot be undone, STORAGE CONFIG {'filepath': '/Users/mbercx/project/core/.aiida/repository/sqlite_dos_7b04381f910f45e086f0e6d0fd7327f0'}
Success: Profile `sqlitedos` was deleted.
```

Reply:
This is just weird because I unintentionally left a debug print in the code. If I remove it, it will look like:
```
❯ verdi profile delete sqlitedos -f --delete-storage
Warning: Deleting profile `sqlitedos`, including the storage.
Warning: This operation cannot be undone, are you sure you want to continue? [y/N]:
Success: Profile `sqlitedos` was deleted.
```
That seems reasonable, doesn't it?

Reply:
Well, it's not really what you show, since the confirmation is only asked when the `-f` is _not_ used. So what I get is:

```
❯ verdi profile delete sqlitedos -f
Warning: Deleting profile `sqlitedos`, without the storage.
Warning: This operation cannot be undone, Success: Profile `sqlitedos` was deleted.
Warning: `sqlitedos` was the default profile, no profiles remain to set as default.
```

Reply:
And _with_ deleting the storage:

```
❯ verdi profile delete sqlitedos -f --delete-storage
Warning: Deleting profile `sqlitedos`, including the storage.
Warning: This operation cannot be undone, Report: Deleted storage directory at `/Users/mbercx/project/core/.aiida/repository/sqlite_dos_cb3f2999582f4ee897a7e3966b7ab43e`.
Success: Profile `sqlitedos` was deleted.
Warning: `sqlitedos` was the default profile, no profiles remain to set as default.
```

Reply:
Ah yeah, of course, I missed the `-f` option. Corrected it now